numpy/core/src/umath/loops.c.src

```diff
@@ -1208,22 +1205,18 @@ NPY_NO_EXPORT void
  * #kind = maximum, minimum#
  * #OP =  >, <#
  **/
-NPY_NO_EXPORT void
+NPY_NO_EXPORT NPY_GCC_OPT_3 void
 @TYPE@_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
 {
-    BINARY_LOOP {
-        const @type@ in1 = *(@type@ *)ip1;
-        const @type@ in2 = *(@type@ *)ip2;
-        if (in1 == NPY_DATETIME_NAT) {
-            *((@type@ *)op1) = in2;
-        }
-        else if (in2 == NPY_DATETIME_NAT) {
-            *((@type@ *)op1) = in1;
-        }
-        else {
-            *((@type@ *)op1) = (in1 @OP@ in2) ? in1 : in2;
-        }
-    }
+    BINARY_LOOP_FAST(@type@, @type@,
+                     if (in1 == NPY_DATETIME_NAT) {
```

Comment:
I don't think we need a fast loop for datetime
does the compiler vectorize this?

Reply:
Nope, with gcc 7.3 and `-ftree-vectorize -fopt-info-vec-missed` you get:
```
numpy/core/src/umath/loops.c.src:1225:5: note: not vectorized: control flow in loop.
numpy/core/src/umath/loops.c.src:1225:5: note: bad loop form.
numpy/core/src/umath/loops.c.src:1225:5: note: not consecutive access n_21 = *dimensions_20(D);
numpy/core/src/umath/loops.c.src:1225:5: note: not vectorized: no grouped stores in basic block.
numpy/core/src/umath/loops.c.src:1225:5: note: not vectorized: not enough data-refs in basic block.
numpy/core/src/umath/loops.c.src:1228:12: note: not consecutive access in1_22 = MEM[(npy_datetime *)ip1_32];
numpy/core/src/umath/loops.c.src:1228:12: note: not consecutive access in2_23 = MEM[(npy_datetime *)ip2_31];
numpy/core/src/umath/loops.c.src:1228:12: note: not vectorized: no grouped stores in basic block.
numpy/core/src/umath/loops.c.src:1229:36: note: not vectorized: not enough data-refs in basic block.
```
Also, the 25% speedup in the asv benchmarks requires both `-O3` and the fast loop.